modules/configuration-annotation-processor/src/main/java/org/apache/ignite/internal/configuration/processor/validation/NamedConfigValueValidator.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.configuration.processor.validation;
+
+import static org.apache.ignite.internal.configuration.processor.ConfigurationProcessorUtils.containsAnyAnnotation;
+import static org.apache.ignite.internal.configuration.processor.ConfigurationProcessorUtils.joinSimpleName;
+
+import javax.annotation.processing.ProcessingEnvironment;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.element.VariableElement;
+import org.apache.ignite.configuration.annotation.Config;
+import org.apache.ignite.configuration.annotation.NamedConfigValue;
+import org.apache.ignite.configuration.annotation.PolymorphicConfig;
+import org.apache.ignite.internal.configuration.processor.ClassWrapper;
+import org.apache.ignite.internal.configuration.processor.ConfigurationValidationException;
+
+/**
+ * Validator for the {@link NamedConfigValue} annotation.
+ */
+public class NamedConfigValueValidator extends Validator {
+    public NamedConfigValueValidator(ProcessingEnvironment processingEnvironment) {
+        super(processingEnvironment);
+    }
+
+    @Override
+    public void validate(ClassWrapper classWrapper) {
```

Comment:
Do you think we should check here that if there is no annotation then we should skip validation?

Reply:
We do:

```
for (VariableElement field : classWrapper.fields()) {
    if (field.getAnnotation(NamedConfigValue.class) == null) {
        continue;
    }
```